src/layers/activations/softmax.cpp

```diff
@@ -75,12 +78,88 @@ void softmax_layer<data_layout::MODEL_PARALLEL, El::Device::CPU>::bp_compute() {
 #ifdef LBANN_HAS_GPU
 template <>
 void softmax_layer<data_layout::MODEL_PARALLEL, El::Device::GPU>::fp_compute() {
-  throw new lbann_exception("Unimplemented method");
+#ifndef LBANN_HAS_CUDNN
+  LBANN_ERROR("cuDNN not detected");
+#else
+  // Local matrices.
+  const auto& local_input = get_local_prev_activations();
+  auto& local_output = get_local_activations();
+  auto& local_workspace = m_workspace->Matrix();
+
+  const El::Int local_height = local_input.Height();
+  const El::Int local_width = local_input.Width();
+
+  // Find the maximum entry in each local column.
+  if (local_width > 0 && local_height == 0) {
```

Comment:
We can simplify these if statements to just `local_height == 0` since the functions in `softmax_cuda` return immediately when `width == 0`.